latte/dataset/dsprites.py

```diff
@@ -1,19 +1,50 @@
 import dataclasses
 import hashlib
 import pathlib
+from enum import IntEnum
+
 import requests
 
 # TODO(Pawel): When pytype starts supporting Literal, remove the comment
-from typing import Literal, Tuple, Union  # pytype: disable=not-supported-yet
+from typing import Literal, Tuple, Union, List, Dict, Optional  # pytype: disable=not-supported-yet
 
 import numpy as np
+from sklearn.preprocessing import StandardScaler
+from sklearn.decomposition import PCA, FactorAnalysis
 
 
 DEFAULT_TARGET = "dataset/raw/dsprites.npz"
 DSPRITES_URL = "https://github.com/deepmind/dsprites-dataset/raw/master/dsprites_ndarray_co1sh3sc6or40x32y32_64x64.npz"
 MD5_CHECKSUM = "7da33b31b13a06f4b04a70402ce90c2e"
 
 
+class DSpritesFactor(IntEnum):
+    COLOR = 0
+    SHAPE = 1
+    SCALE = 2
+    ORIENTATION = 3
+    X = 4
+    Y = 5
+
+
+factor_names = {
+    DSpritesFactor.SHAPE: "Shape",
+    DSpritesFactor.SCALE: "Scale",
+    DSpritesFactor.ORIENTATION: "Orientation",
+    DSpritesFactor.X: "Position x",
+    DSpritesFactor.Y: "Position y",
+}
+
+
```

Comment:
I now know what it does, but it would be good to add a one-line comment :slightly_smiling_face: 
(As we're likely to forget it in 3-month time and checking the code requires 2 minutes instead of 2 seconds :wink: )